### PR TITLE
Use BOM and dependabot

### PR DESCRIPTION
**Pull request by warden**

### CODE REVIEW

File: pom.xml
Comment (jglick) [inline pom.xml:37]: Did you use `mvn incrementals:incrementalify`? You forgot to commit the `.mvn/` directory if so.

File: pom.xml
Comment (jglick) [inline pom.xml:66]: Copy-paste? Use the Maven tool.

```suggestion
    <revision>1.21</revision>
```

File: pom.xml
Comment (jglick) [inline pom.xml:182]: Delete this dep. It should be picked up from `structs`, never used as a direct dep.

File: pom.xml
Comment (jglick) [inline pom.xml:209]: Revert, this is critical.

**Review by jglick — CHANGES_REQUESTED** [review]
Several mistakes.

File: pom.xml
Comment (warden) [reply pom.xml:209]: Why is it critical (so I can put a comment above that it should not be touched) ?
Also, shouldn't below be ${java.level} instead of 1.5?

Comment (jglick) [reply pom.xml:209]: > Why is it critical

Because of one of the libraries it is using, SSHD IIRC. Without `pluginFirstClassLoader` that mode of the plugin will just throw errors. https://www.jenkins.io/doc/developer/plugin-development/dependencies-and-class-loading/

`compatibleSinceVersion` should also stay, though at this point it is mostly of historical interest since it is unlikely people would be updating since such an old version.

File: pom.xml
Comment (jglick) [inline pom.xml:188]:
```suggestion
```

Nobody does this. https://github.com/jenkinsci/plugin-pom/pull/133 may do this automatically some day.

Comment (warden) [reply pom.xml:188]: This is added for compliancy with XSD as suggested by my intelliJ as a required param in `<configuration>`.

Comment (jglick) [reply pom.xml:190]: Well tell your IDE to shut up. :grin:

Comment (jglick) [reply pom.xml]: https://github.com/jenkinsci/plugin-pom/blob/0c6a61e618eaa913bd0c8489d12a58c78f0bdfed/pom.xml#L644 already defines this, so IDEA is presumably just not taking inheritance into account.

File: .github/dependabot.yml
Comment (jglick) [inline .github/dependabot.yml:9]: Why is this here? This plugin’s `pom.xml` makes no reference to Guava. I think this whole `ignore` section can be deleted.

File: pom.xml
Comment (jglick) [inline pom.xml:126]: Could be deleted; would be picked up transitively anyway.

File: pom.xml
Comment (jglick) [inline pom.xml:151]: ditto

File: pom.xml
Comment (jglick) [inline pom.xml:157]: (but this needs to stay, if we are in fact using these utilities)

**Review by jglick — APPROVED** [review]

File: pom.xml
Comment (jglick) [inline pom.xml:139]: You probably did need this.

File: pom.xml
Comment (jglick) [inline pom.xml:145]: And this.

(Unless both are getting picked up from `docker-workflow`?)

Comment (warden) [reply pom.xml:145]: Tests are compiling fine and passing, why would I need this then?

Comment (jglick) [reply pom.xml:145]: Probably it is getting picked up from `docker-workflow` then.

**Comment by warden** [comment]
@jglick thanks for help on this.  What's next? 

Iis the `<maintainers>` pom field still used across plugins and up2date in this case?  Looking at the merge history looking at the people who merged, it looks to me that there is a de-sync between maintainers and actual people that have repo merge rights.
After all, CI passed and there is an approval from you (so one of the people having merge rights) - wouldn't that work if in such case an auto-merge is done by the pipeline?

As a contributor, I'm just wondering if we can speed up the PR merging process somehow as it's not clear what happens here next.

**Comment by jglick** [comment]
POM `maintainers` can be ignored. Maybe @MRamonLeon is maintaining these days?

**Comment by warden** [comment]
@MRamonLeon any thoughts?

**Comment by warden** [comment]
@MRamonLeon, @jglick ping again 🙏

**Review by MRamonLeon — APPROVED** [review]

**Comment by MRamonLeon** [comment]
I don't see any reason to release these changes. Do you need it for any reason? Thank you for your contribution @warden

**Comment by warden** [comment]
@MRamonLeon thanks for acting on this. No, this one is not needed to be released, more important is #30.

**Comment by MRamonLeon** [comment]
I've commented on #30 @warden

File: pom.xml
Comment (jglick) [reply pom.xml:37]: Still missing `.mvn/extensions.xml`.

Comment (rantoniuk) [reply pom.xml:37]: Feel free to add that one file instead of commenting 2y old already merged PR ;-)
At the same time, might be just as well good to consider adding this file globally to all repositories under /jenkinsci org - I don't think it would harm them even if they're not using incrementals at the moment?

Comment (jglick) [reply pom.xml:37]: > Feel free to add that one file

Already done in #85.

> consider adding this file globally to all repositories

Could, but probably more confusing than helpful, since `mvn incrementals:incrementalify` expects to add it.

**Pull request merged**
